### DISCUSSION TITLE
I bought my first pellet stove and want to control it with an ESP32

### DISCUSSION THREAD

**Discussion by Geo-Ron** [jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip#37] [General]
Hi Jeff.

I bought my first pellet stove. Just the same hardware as yours, or so it seems.

So I am starting the journey to connect it to HomeAssistant. This without having a wifi-board on it.

I am publishing my progress and information at my repository at https://github.com/Geo-Ron/esphome-PelletStove and will be reporting back if I can.

My first job it to understand how you actually wired up everything and what is happening.

I really liked reading your posts! Fantastic work! You are the best!

**Comment by jazzmonger** [comment]
Awesome!! It's a fun and rewarding journey!Assuming your control unit internals look like mine, it *should* work.  

any idea WHAT APP is used to control your stove?? Blynk?  Tuya?  Tuya is what we want....

Be warned that some of the EU models use a different controller. I have not had anyone willing to experiment past reflashing the TYWE1S chips on those (which didn't work).Step 1: get a full blown esp32, not a c3. It has to say ESP32-WROOM on it.  I use the mini version as it's small and has ever needed.This will set you up for anything you could want to do in the future.  The S2 & C3 and other variants just don't implement all the UART features fully. You'll want to start out with my basic config. i.e. replacing the (non-existent) wifi module with the esp32 and flashing the basic control code to that.It's pretty straightforward.  If you get stuck, just ask me.  As you can see, I have been literally obsessed with these pellet stoves for a few years now, but I've completely tamed them and it's a great feeling to rejoice in defeating shoddy programmers.

Jeff

start here: https://github.com/jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip/discussions/33

**Comment by jazzmonger** [comment]
Yup, after looking closely at your controller is exactly like mine so it should work...

**Comment by Geo-Ron** [comment]
Ha!

After reading the discussions, I think I understand how it works.

> What I do not fully understand is how it should be wired to the MCU/Display .... 😄
> And how the wires lead on the display pcb

What I currently think I understand: 

- TX from J1 (display <-> mcu-connection) is rerouted to J5 - headerpin 8, using R18 pad (modification)
  - this is called 'display UART' and is READ by UART2
- RX from J1 (display <-> mcu-connection) is retouted to J5 - headerpin 6, using R19 pad (modification)
  - this is called 'display UART' and is READ by UART3
- TX from TuyaMCU is wired to J5 - headerpin 12 (display PCB original)
  - this is called 'TuyaMCU UART' and is READ by UART1
- TX from TuyaMCU is wired to J5 - headerpin 10 (display PCB original)
  - this is called 'TuyaMCU UART' and is TRANSMITTING by UART1
- LED positive is rewired to J5 - headerpin 9 (modification, as shown in [this image](https://user-images.githubusercontent.com/52110065/221762837-a2237990-73e0-4747-bd84-9496e9e10223.jpeg) or [this image](https://user-images.githubusercontent.com/52110065/222992725-f3061c77-b96f-4f97-8a80-39b15f2e8031.jpeg)  )
- Headerpin 15 is reset?? this is unknown to me.
  - is this the 'Thumbs Up - press'? How is it wired?

> Much of this is derived from ALL the discussions, combined with [this post](https://github.com/jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip/discussions/34#discussioncomment-7350290)

## Wiring diagram as I have understood things...

![image](https://github.com/user-attachments/assets/5aaa59e3-be3c-4284-a001-86d4975f3c7b)

**Reply by jazzmonger** [reply]
nice - I hadn't used KiCad until recently & should have drawn this out long ago. At one point I was going to make a circuit board but never got around to it.  I've since made several (voltage regulators for the Ford model T, which I now sell)  but none with ESPs yet.

> Headerpin 15 is reset?? this is unknown to me.
> * is this the 'Thumbs Up - press'? How is it wired?

Wired to the thumbs up button on the side of the switch that isn't grounded.  When the stove fails (ie fire goes out or it throws an error code) this lets you reset the stove remotely with a toggle in HA instead of physically having to do it on the front panel. It's one of the first things I should have done but ended up as one of the last.  VERY handy as you'll find out.

**Comment by jazzmonger** [comment]
Trace your wires and take a look here:
https://github.com/jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip/discussions/34#discussioncomment-7352643

**Comment by Geo-Ron** [comment]
Next week I get physical access to the pellet stove. So then I can trace the wires.

So far I've build a [KiCad schematic](https://kicanvas.org/?github=https%3A%2F%2Fgithub.com%2FGeo-Ron%2Fesphome-PelletStove%2Ftree%2Fmain%2Fkicad%2Fesphome-pelletkachel)

This should do it. Eventually will try to create a PCB for it.

**Comment by jazzmonger** [comment]
Looks fantastic - I don't see any issues.  I like your interconnect idea.
<img width="251" alt="image" src="https://github.com/user-attachments/assets/1e5d2b18-525b-4ddf-94db-568540b688bf">
 You're taking it to the next level.  Keep me posted if you do get boards made - I'd be willing to help w/ that.  JLPCB has a minimum order qty I believe - I'd take 3-4 of them.

Jeff

**Comment by Geo-Ron** [comment]
I need to measure the distances on the connections for the display pcb and the available space in the box.
But the basics are ready.

PCB in KiCad prepared, see [schematic](https://kicanvas.org/?github=https%3A%2F%2Fgithub.com%2FGeo-Ron%2Fesphome-PelletStove%2Ftree%2Fmain%2Fkicad%2Fesphome-pelletkachel)

Now on to the breadboard...

**Reply by jazzmonger** [reply]
you might want to rethink w/ a D1 Mini 32 rather than a full size ESP32 Dev board.  There's just not that much space in there.... thats the only ESP I could get to fit.
